src/Symfony/Component/Config/Definition/BooleanNode.php

```diff
@@ -20,10 +20,23 @@
  */
 class BooleanNode extends ScalarNode
 {
+    public function __construct(
+        ?string $name,
+        ?NodeInterface $parent = null,
+        string $pathSeparator = self::DEFAULT_PATH_SEPARATOR,
+        protected bool $nullable = false,
```

Comment:
why protected?

Reply:
`BooleanNode` is not final and children classes could benefit from this. But let's make it private for less maintenance cost and change visibility one day if there's a need